DropboxSync.UIL/Program.cs

```diff
@@ -1,2 +1,16 @@
-﻿
-Console.WriteLine();
\ No newline at end of file
+﻿using DropboxSync.UIL;
+using Microsoft.Extensions.DependencyInjection;
+
+StartUp.Build();
+
+Console.ForegroundColor = ConsoleColor.Blue;
+Console.WriteLine("Dropbox synchronisation started");
+Console.ForegroundColor = ConsoleColor.Gray;
+
+BrokerEventListener brokerEventListener = StartUp.ServiceProvider?.GetService<BrokerEventListener>() ??
+    throw new NullReferenceException(nameof(BrokerEventListener));
+
+brokerEventListener.Initialize();
+brokerEventListener.Start();
+
+Console.ReadKey();
```

Comment:
this seems a better solution => https://medium.com/@rainer_8955/gracefully-shutdown-c-apps-2e9711215f6d

Reply:
As I checked further what you said, it is indeed not just a better solution but is the best one.

So I deleted `StartUp.cs` and rewrote the `Program.cs` to fit the right way

`Program.cs`
```csharp
using DropboxSync.UIL;
using DropboxSync.UIL.Managers;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Microsoft.Extensions.Logging;

var host = new HostBuilder()
    .ConfigureLogging(logging =>
    {
        logging.AddConsole();
    })
    .ConfigureServices(services =>
    {
        services.AddHostedService<ShutdownManager>();
        services.AddTransient<IExpenseManager, ExpenseManager>();
        services.AddTransient<IInvoiceManager, InvoiceManager>();
        services.AddTransient<IDossierManager, DossierManager>();
        services.AddSingleton<BrokerEventListener>();
    })
    .UseConsoleLifetime()
    .Build();

await host.RunAsync();
```

The app doesn't wait for a key being pressed now but instead the combination `CTRL + C`

On `CTRL + C` press
![image](https://user-images.githubusercontent.com/56565073/167400765-a1465efc-d64a-46cd-9da9-d08f4b798070.png)

But this is not the only difference. You can remark in `ConfigureLogging(loggin => ...)` I added `Logging.AddConsole()` so from now on the logging in the console looks like this

After app startup and Invoice Event receival
![image](https://user-images.githubusercontent.com/56565073/167400604-bf5fe8d2-f1b8-4a37-9542-fefd781c9bfa.png)